### PR TITLE
Upgrade to mongo 3.4 and added volume driver choice

### PR DESCRIPTION
**Pull request by joshuacox**
I also threw a scale choice in the questions

### CODE REVIEW

File: templates/MongoDB/3/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      CATTLE_SCRIPT_DEBUG: ${debug}
+    entrypoint: /opt/rancher/bin/entrypoint.sh
+    command:
+    - --replSet
+    - "${replset_name}"
+    image: mongo:3.4
+    labels:
+      io.rancher.scheduler.affinity:host_label: ${host_label}
+      io.rancher.container.hostname_override: container_name
+      io.rancher.sidekicks: mongo-base, mongo-datavolume
```
Comment (rawmind0) [inline templates/MongoDB/3/docker-compose.yml]: Affinity rule to distribute instances by hosts, preventing run more than one by host.
```
+      io.rancher.scheduler.affinity:container_label_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
```

File: templates/MongoDB/3/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - mongo-datavolume
+      - mongo-base
+    ports:
+      - 27017:27017/tcp
+  mongo-base:
+    restart: always
+    labels:
+      io.rancher.scheduler.affinity:host_label: ${host_label}
+      io.rancher.container.hostname_override: container_name
+      io.rancher.container.start_once: true
```
Comment (rawmind0) [inline templates/MongoDB/3/docker-compose.yml]: Affinity rule to distribute instances by hosts, preventing run more than one by host.
```
+      io.rancher.scheduler.affinity:container_label_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
```

File: templates/MongoDB/3/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    image: rancher/mongodb-conf:v0.1.1
+    stdin_open: true
+    entrypoint: /bin/true
+  mongo-datavolume:
+    labels:
+      io.rancher.scheduler.affinity:host_label: ${host_label}
+      io.rancher.container.hostname_override: container_name
+      io.rancher.container.start_once: true
```
Comment (rawmind0) [inline templates/MongoDB/3/docker-compose.yml]: Affinity rule to distribute instances by hosts, preventing run more than one by host.
```
+      io.rancher.scheduler.affinity:container_label_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
```

**Comment by rawmind0** [comment]
Hi @joshuacox .. thanks so much for the PR....

Package is working, but just one change to get better. Please, take a look the change request. :)

**Comment by joshuacox** [comment]
Is that correct?

**Comment by rawmind0** [comment]
Cool, thanks....

LGTM

**Pull request merged**